docs/sources/clients/promtail/scraping.md

```diff
@@ -206,6 +206,10 @@ Before using `gcplog` target, GCP should be [configured](../gcplog-cloud) with p
 
 It also support `relabeling` and `pipeline` stages just like other targets.
 
+Log entries scraped by `gcplog` will add additional label called `promtail_instance`. This label uniquely identifies each promtail instances trying to scrape gcplog(from single `subscription_id`).
+We need this unique identifier to avoid out-of-order errors from Loki servers.
+Because say two promtail instances rewrite timestamp of log entries(with same labelset) at the same time may reach Loki servers at different times can cause Loki servers to reject it.
```

Comment:
I think here you should perhaps explain a bit more what causes this conflict in log lines - it may not be immediately obvious to users who are new to this. You've added a config option [in the docs](https://github.com/grafana/loki/blob/c5b7fc1841675048b9eec8ee3c5cf32702f90fb1/docs/sources/clients/promtail/scraping.md#gcplog-scraping) (`use_incoming_timestamp: false # default rewrite timestamps`) but this is optional - correct? If users don't notice that log line, they may be confused as to why we are rewriting the timestamp.

This adds another wrinkle: if users opt to not rewrite the timestamp, is the unique UUID even required? (we can leave it in either way, just curious)

Reply:
Regarding the wrinkle: I tried with and without rewriting the timestamp during my deployment. In both cases `out of order` errors appeared, so this UUID is still needed to be able to ingest GCP logs.

Maybe this is not needed when you are limiting the amount of logs, but I already had issues with just the `k8s_cluster` resource logs.